src/org/omegat/filters2/moodlephp/MoodlePHPFilter.java

```diff
@@ -0,0 +1,218 @@
+/**************************************************************************
+ OmegaT - Computer Assisted Translation (CAT) tool
+          with fuzzy matching, translation memory, keyword search,
+          glossaries, and translation leveraging into updated projects.
+
+ Copyright (C) 2018 Enrique Estevez Fernandez
+               Home page: http://www.omegat.org/
+               Support center: http://groups.yahoo.com/group/OmegaT/
+
+ This file is part of OmegaT.
+
+ OmegaT is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ OmegaT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+     **************************************************************************/
+
+package org.omegat.filters2.moodlephp;
+
+import java.awt.Window;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.omegat.filters2.AbstractFilter;
+import org.omegat.filters2.FilterContext;
+import org.omegat.filters2.Instance;
+import org.omegat.filters2.TranslationException;
+import org.omegat.util.Log;
+import org.omegat.util.NullBufferedWriter;
+import org.omegat.util.OStrings;
+import org.omegat.util.StringUtil;
+
+/**
+ * Filter for support Moodle PHP files.
+ * 
+ * Code adapted from the file: MozillaDTDFilter.java
+ *
+ * @author Enrique Estevez (keko.gl@gmail.com)
+ */
+public class MoodlePHPFilter extends AbstractFilter {
+
+    public static final String OPTION_REMOVE_STRINGS_UNTRANSLATED = "unremoveStringsUntranslated";
+
+    protected static final Pattern RE_ENTITY = Pattern.compile("\\$string\\['(.+)'] (=) '(.+)';",
+            Pattern.DOTALL);
+
+    protected Map<String, String> align;
+
+    /**
+     * If true, will remove non-translated segments in the target files
+     */
+    public static boolean removeStringsUntranslated = false;
```

Comment:
I know you based this on the Mozilla DTD filter, and this part is the same as in that one, but it doesn't make any sense for this to be `static`. Can you please make it non-static?

Reply:
Changed it.